spec/grape/jsonapi/parameters/filter_spec.rb

```diff
@@ -142,26 +143,26 @@
     context 'eq and in' do
       let(:json) { JSON.unparse(aaa: 123, bbb: [2, 3, 4]) }
       before do
-        allow(model).to receive(:where).and_return(model)
-        allow(model).to receive(:in).and_return(model)
+        allow(model).to receive(:where).and_return(query)
+        allow(query).to receive(:in).and_return(query)
       end
       it 'it calls mutators on the model' do
-        expect(subject).to eq model
+        expect(subject).to eq query
         expect(model).to have_received(:where).with(aaa: 123)
-        expect(model).to have_received(:in).with(bbb: [2, 3, 4])
+        expect(query).to have_received(:in).with(bbb: [2, 3, 4])
       end
     end
 
     context 'gte and ne' do
       let(:json) { JSON.unparse(aaa: { gte: 123 }, bbb: { ne: 44 }) }
       before do
-        allow(model).to receive(:gte).and_return(model)
-        allow(model).to receive(:ne).and_return(model)
```

Comment:
for example these allows are order dependent.  which i feel like is an overfit spec.